pkg/rpcclient/nns/contract.go

```diff
@@ -120,3 +160,235 @@ func itemsToRecords(arr []stackitem.Item) ([]RecordState, error) {
 	}
 	return res, nil
 }
+
+// Roots invokes `roots` method of contract.
+func (c *ContractReader) Roots() (uuid.UUID, result.Iterator, error) {
+	return unwrap.SessionIterator(c.invoker.Call(c.hash, "roots"))
+}
+
+// RootsExpanded is similar to Roots (uses the same contract
+// method), but can be useful if the server used doesn't support sessions and
+// doesn't expand iterators. It creates a script that will get the specified
+// number of result items from the iterator right in the VM and return them to
+// you. It's only limited by VM stack and GAS available for RPC invocations.
+func (c *ContractReader) RootsExpanded(_numOfIteratorItems int) ([]RecordState, error) {
+	arr, err := unwrap.Array(c.invoker.CallAndExpandIterator(c.hash, "roots", _numOfIteratorItems))
+	if err != nil {
+		return nil, err
+	}
+	return itemsToRecords(arr)
```

Comment:
NACK. Have you ever tried to invoke this method on real NNS in testnet? Try, and it will fail, because the returned value of `roots` is array of strings.